neo/rawio/openephysbinaryrawio.py

```diff
@@ -191,12 +223,30 @@ def _parse_header(self):
                         raise ValueError(
                             "SYNC channel is not present in the recording. " "Set load_sync_channel to False"
                         )
-
-                    if has_sync_trace and not self.load_sync_channel:
-                        self._stream_buffer_slice[stream_id] = slice(None, -1)
+                    
+                    num_neural_channels = sum(1 for ch in info["channels"] if "ADC" not in ch["channel_name"])
+                    num_non_neural_channels = sum(1 for ch in info["channels"] if "ADC" in ch["channel_name"])
+                    
+    
+                    if num_non_neural_channels == 0:                        
+                        if has_sync_trace and not self.load_sync_channel:
+                            self._stream_buffer_slice[stream_id] = slice(None, -1)
+                        else:
+                            self._stream_buffer_slice[stream_id] = None
                     else:
-                        self._stream_buffer_slice[stream_id] = None
-
+                        # For ADC channels, we remove the last channel which is the Synch channel
+                        stream_id_neural = stream_id
+                        stream_id_non_neural = str(int(stream_id) + self._num_of_signal_streams)
+                        
+                        # Note this implementation assumes that the neural channels come before the non-neural channels
```

Comment:
Here guys, these are the strongest assumptions that the reader makes. I wanted to check with @alejoe91 if he knows about this:
* Is it true that the synch trace will always be the last channel even if you have non-neural channels? I could not find this in the documentation.
* Are the neural channels and non-neural always in order and not interleaved? 
 
 If the first case is not True, we can extract the specific index from the structure.oebin. That's a bigger change but I think doable.
 
 For the second case, if it is not true, I am afraid we will have to error somehow as the current data model does not support non-contigious streams across the buffer with the buffer slice mechanism (@samuelgarcia )

Reply:
I am adding error messages for this advising people to open an issue if they have interleaved neural and non-neural or synch trace that is not at the end.